flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/MySqlValidator.java

```diff
@@ -66,7 +66,9 @@ public void validate() {
                 connection = DebeziumUtils.openJdbcConnection(sourceConfig);
             } else {
                 // for the legacy source
-                connection = DebeziumUtils.createMySqlConnection(from(dbzProperties));
+                connection =
+                        DebeziumUtils.createMySqlConnection(
+                                from(dbzProperties), sourceConfig.getJdbcProperties());
```

Comment:
```suggestion
                        DebeziumUtils.createMySqlConnection(sourceConfig);
```

Reply:
This is used by the legacy source whose `sourceConfig` is null. I will revert this change to use `new Properties()`.